src/Authoring/WinRT.SourceGenerator/WinRTRules.cs

```diff
@@ -21,121 +21,130 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
                 helpLinkUri: "https://docs.microsoft.com/en-us/previous-versions/hh977010(v=vs.110)");
         }
 
+        public static DiagnosticDescriptor ImplementerUseSameVariableName = MakeRule(
+            "CsWinRT1113", 
+            "Class implementaiton method should use same paramter as interface method",
+            "Class '{0}' implements method '{1}' with parameter '{2}', but '{1}' was declared in interface '{3}' with parameter '{4}'." 
+            + "Parameter names must match exactly in the Windows Runtime.");
+        
         public static DiagnosticDescriptor PrivateGetterRule = MakeRule(
-            "WME", 
+            "CsWinRT", 
             "Property must have public getter",
-            "Property '{0}' does not have a public getter method. Windows Metadata does not support setter-only properties.");
+            "Property '{0}' does not have a public getter method. Windows Runtime does not support setter-only properties.");
         
         public static DiagnosticDescriptor DisjointNamespaceRule = MakeRule(
-            "WME1044",
+            "CsWinRT1044",
             "Namespace is disjoint from main (winmd) namespace",
             "A public type has a namespace ('{1}') that shares no common prefix with other namespaces ('{0}'). "
             + "All types within a Windows Metadata file must exist in a sub namespace of the namespace that is "
             + "implied by the file name.");
 
         public static DiagnosticDescriptor NamespacesDifferByCase = MakeRule(
-            "WME1067",
+            "CsWinRT1067",
             "Namespace names cannot differ only by case",
-            "Multiple namespaces found with the name '{0}', namespace names cannot differ only by case.");
+            "Multiple namespaces found with the name '{0}'; namespace names cannot differ only by case in the Windows Runtime.");
 
         public static DiagnosticDescriptor NoPublicTypesRule = MakeRule(
-            "WME1042",
+            "CsWinRT1042",
             "No public types defined",
             "Windows Runtime components must have at least one public type");
 
         public static DiagnosticDescriptor GenericTypeRule = MakeRule(
-            "WME",
+            "CsWinRT",
             "Class (or interface) is generic",
             "Type {0} is generic. Windows Runtime types cannot be generic.");
 
 
         public static DiagnosticDescriptor UnsealedClassRule = MakeRule(
-            "WME",
+            "CsWinRT",
             "Class is unsealed",
-            "Exporting unsealed types is not supported. Please mark type {0} as sealed.");
+            "Exporting unsealed types is not supported in Windows Runtime. Please mark type {0} as sealed.");
```

Comment:
I would remove the in Windows Runtime as technically it is now supported in WinRT, but we still have that limitation.

Reply:
Do we want to remove the diagnostic for this then?

Reply:
CsWinRT still has the limitation that requires it to be sealed.

Reply:
Per discussion, will change to CsWinRT.